toonz/sources/toonz/shortcutpopup.cpp

```diff
@@ -415,4 +471,320 @@ void ShortcutPopup::onSearchTextChanged(const QString &text) {
 
 //-----------------------------------------------------------------------------
 
+void ShortcutPopup::onPresetChanged(int index) {
+  if (m_presetChoiceCB->currentText() == "Load from file...") {
+    importPreset();
+  }
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::clearAllShortcuts(bool warning) {
+  if (warning) {
+    QString question(tr("This will erase ALL shortcuts. Continue?"));
+    int ret =
+        DVGui::MsgBox(question, QObject::tr("OK"), QObject::tr("Cancel"), 0);
+    if (ret == 0 || ret == 2) {
+      // cancel (or closed message box window)
+      return;
+    }
+    showDialog("Clearing All Shortcuts");
+  }
+  for (int commandType = UndefinedCommandType; commandType <= MenuCommandType;
+       commandType++) {
+    std::vector<QAction *> actions;
+    CommandManager::instance()->getActions((CommandType)commandType, actions);
+    for (QAction *action : actions) {
+      qApp->processEvents();
+      m_sViewer->setAction(action);
+      m_sViewer->removeShortcut();
+    }
+  }
+  setCurrentPresetPref("DELETED");
+  // if warning is true, this was called directly- need to hide the dialog after
+  if (warning) m_dialog->hide();
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::setPresetShortcuts(TFilePath fp) {
+  QSettings preset(toQString(fp), QSettings::IniFormat);
+  preset.beginGroup("shortcuts");
+  QStringList allIds = preset.allKeys();
+  QAction *action;
+  for (QString id : allIds) {
+    QByteArray ba      = id.toLatin1();
+    const char *charId = ba.data();
+    action = CommandManager::instance()->getAction((CommandId)charId);
+    CommandManager::instance()->setShortcut(
+        action, preset.value(id).toString().toStdString(), false);
+  }
+  preset.endGroup();
+  m_sViewer->shortcutChanged();
```

Comment:
Please add `emit` before `m_sViewer->shortcutChanged();` in order to make it easier to understand that this line calls signal... Or you can call `m_list->onShortcutChanged()` directly since for now this signal is only for updating ShortcutTree.